runtime/tests/staking_reward.rs

```diff
@@ -271,42 +253,36 @@ fn elected_validators_receive_equal_transaction_fee_reward() {
 			}
 
 			start_era(2);
-			// Start the first block of era 2
-			let header = header_for_block_number((System::block_number() + 1).into());
-			Executive::initialize_block(&header);
+			Executive::initialize_block(&header_for_block_number((System::block_number() + 1).into()));
+
+			// The whole reward of the era goes to the author
+			let author_reward = Rewards::target_inflation_per_staking_era();
 
-			// Check if stash account balances are not yet changed
-			let per_validator_reward_era_2: Balance = ((Perbill::one()
-				.saturating_sub(Rewards::development_fund_take()))
-				* tx_fee + Rewards::target_inflation_per_staking_era())
-				/ validators.len() as Balance;
 			for (stash, _controller, _, _, _, _) in &validators {
-				assert_eq!(
-					RewardCurrency::free_balance(&stash),
-					initial_balance + per_validator_reward_era_1 + per_validator_reward_era_2
-				);
+				if stash == &author_stash_id {
+					assert_eq!(
+						RewardCurrency::free_balance(&stash),
+						initial_balance + per_validator_reward_era_1 + author_reward
+					);
+				} else {
+					assert_eq!(
+						RewardCurrency::free_balance(&stash),
+						initial_balance + per_validator_reward_era_1
+					);
+				}
 			}
 
-			// treasury gets it's cut
```

Comment:
should still check treasury cut here

Reply:
With the new arrangement of this test, the tx_fee is zero. I can add an individual test for the treasury cut and keep this one simple. I think simpler single-purpose tests are better maintainable and can help us more. So I prefer adding another test personally. What's your opionion?

Reply:
I'd say keep the original behaviour of the test and check authorship points also

Reply:
The problem I had in this test was stemmed from Executive::apply_extrinsic not triggering note_author. So as said above, I'm going to add a separate test for this similar to the previous test here which is not checking for points at the same time.